lib/CMakeLists.txt

```diff
@@ -10,52 +10,44 @@ configure_file(
     @ONLY
 )
 
-# Files to be compiled into the library
-add_library(mxl SHARED)
-target_compile_features(mxl
-        PRIVATE
-            cxx_std_20
-    )
-set_target_properties(mxl
-        PROPERTIES
-            POSITION_INDEPENDENT_CODE    ON
-            VISIBILITY_INLINES_HIDDEN    ON
-            C_VISIBILITY_PRESET          hidden
-            CXX_VISIBILITY_PRESET        hidden
-            C_EXTENSIONS                 OFF
-            CXX_EXTENSIONS               OFF
-            VERSION                      "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}"
-            SOVERSION                    "${PROJECT_VERSION_MAJOR}"
-    )
+add_library(mxl-objects OBJECT
+    src/flow.cpp
+    src/mxl.cpp
+    src/time.cpp
```

Comment:
I think these three are conceptually `PRIVATE` to `mxl` rather than being part of `mxl-objects`.